### ISSUE TITLE
[3.4] Segmentation Evaluation Framework

### ISSUE THREAD

**Issue by poltergeistjoya** [poltergeistjoya/Vacant_Lot_Detection#20]
## Summary
Implement comprehensive evaluation metrics and visualization functions for segmentation results. Can be implemented in parallel with 3.1/3.2/3.3.

### Add to `vacant_lot/segmentation.py`

```python
def evaluate_segmentation(model, test_dataset, device, threshold=0.5):
    '''Compute metrics on test set (Brooklyn+Queens)'''
    return {
        'iou': Intersection over Union,
        'dice': F1-like metric,
        'precision': TP / (TP + FP),
        'recall': TP / (TP + FN),
        'f1': 2 * (precision * recall) / (precision + recall),
        'kappa': Cohen's Kappa,
        'pr_curve': (thresholds, precisions, recalls),
        'per_patch_iou': [IoU for each patch],
        'confusion_matrix': [[TN, FP], [FN, TP]],
    }

def visualize_predictions(model, dataset, indices, output_dir):
    '''Generate RGB | GT | Pred | Overlay visualizations'''
    for idx in indices:
        image, mask = dataset[idx]
        pred = model(image.unsqueeze(0))
        fig = plot_comparison(image, mask, pred)
        save_fig(f'{output_dir}/patch_{idx}.png')

def plot_iou_distribution(per_patch_ious, output_path):
    '''Histogram: per-patch IoU distribution'''

def plot_confusion_matrix(confusion_matrix, output_path):
    '''Heatmap: TN/FP/FN/TP normalized'''

def plot_pr_curve(precisions, recalls, output_path):
    '''Precision-Recall curve with AP'''

def plot_pixel_distributions(preds, targets, output_path):
    '''Histogram: predicted probability distribution by class'''
```

## Metrics Definition

| Metric | Formula | Interpretation |
|--------|---------|-----------------|
| **IoU** | TP / (TP+FP+FN) | Primary metric; target area overlap |
| **Dice/F1** | 2×TP / (2×TP+FP+FN) | Similar to IoU; emphasizes recall |
| **Precision** | TP / (TP+FP) | "Of flagged pixels, how many are truly vacant?" |
| **Recall** | TP / (TP+FN) | "Of truly vacant pixels, how many did we find?" |
| **Kappa** | (po - pe) / (1 - pe) | Agreement beyond chance; handles imbalance |
| **AP (PR-AUC)** | Area under Precision-Recall curve | Useful for imbalanced classes |

## Avoid
- **Overall Accuracy:** Misleadingly high with imbalanced data (99% accuracy if model predicts all-negative)
- **Threshold optimization on test set:** Bias. Fix threshold=0.5 on test; optimize on val only.

## Acceptance Criteria
- [ ] `evaluate_segmentation()` computes all 6 metrics correctly
- [ ] Visualization functions generate high-quality PNG/PDF figures
- [ ] Per-patch IoU histogram shows distribution across patches
- [ ] PR curve computes correctly (AP matches sklearn if compared)
- [ ] Confusion matrix normalized and readable
- [ ] Tested on dummy model + dataset; results make sense

## Usage in Week 4
- Called on test set (Brooklyn+Queens) in issue 4.2
- Used in results notebook (issue 4.3) for final comparison table

## Dependencies
- sklearn.metrics (for baseline implementations)
- matplotlib, seaborn (for visualization)

## References
- Müller et al. (2022), Towards a guideline for evaluation metrics
- Kaggle competition metrics guide
